src/main.rs

```diff
@@ -15,10 +41,12 @@ fn read_from_source<T: BufRead>(reader: &mut T) -> Value {
         }
     };
 
+    // empty file error
```

Comment:
This is actually not an error. The file/stdin is empty and there's just nothing to do, so just exit (nonerror status 0). At least that's how I was handling it.

On the other hand, if you executed `jp_cli -f input.json some.path` given the following assumptions

1. input.json is an empty file
2. some.path query was given
3. invalid query is an error

Which should take precedence? The file being empty (therefore ignore query and exit with zero status) or the query being invalid? I guess the former is probably where I lean. Please share any thoughts you have on this.

Reply:
Ah, that makes much more sense than what I was thinking. 

Can you just remove the `if size == 0` check? I believe that would fall back to a `JsonParseError`, which seems reasonable to me. Let me know what you'd like to do and I'll update the PR.

Reply:
I was doing that because the app was panicking if I didn't handle it. I'll play around with the code again to see if it was more to do with how the code was structured before all my refactoring.

Reply:
I just checked the behavior on my branch. As is, with an empty `input.json`, it returns `Error: EmptyFileError`. If I remove the `if size == 0` check, it returns `Error: JsonParseError`.

It seems like if serde is going to throw an error anyway, you might as well return a more specific error letting the user know their input was empty.

Reply:
For now let's leave it in until I decide how I want to handle it.